hub/core/chunk_engine.py

```diff
@@ -137,15 +143,21 @@ def chunk_id_encoder(self) -> ChunkIdEncoder:
                 )
 
             enc = ChunkIdEncoder()
-            self.cache[key] = enc
+            cache[key] = enc
             return enc
 
-        enc = self.cache.get_cachable(key, ChunkIdEncoder)
+        enc = cache.get_cachable(key, ChunkIdEncoder)
         return enc
 
     @property
     def chunk_id_encoder_exists(self) -> bool:
-        return get_chunk_id_encoder_key(self.key) in self.cache
+        cache = self.mem_cache or self.cache
+        try:
+            key = get_chunk_id_encoder_key(self.key)
```

Comment:
this should be used everywhere to check existence as listing all s3 items is an expensive operation and in current implementation s3provider lists only 1000 items